docs/form.md

```diff
@@ -0,0 +1,93 @@
+# Having components participate in form
+
+Though form elements in `carbon-custom-elements`, e.g. `<bx-input>` are not native form elements like `<input>`, form elements in `carbon-custom-elements` have some extra APIs that align well to web/framework standards that allows those form elements to participate in form.
+
+## `formdata` event
+
+Browsers supporting [`formdata` event](https://docs.google.com/document/d/1JO8puctCSpW-ZYGU8lF-h4FWRIDQNDVexzHoOQ2iQmY/edit#heading=h.je8c7y5qpgki) fires that event when user clicks on `<button type="submit">` in `<form>`. Our form components listen to that event to add their values to the `<form>`.
```

Comment:
```suggestion
Browsers supporting [`formdata` event](https://docs.google.com/document/d/1JO8puctCSpW-ZYGU8lF-h4FWRIDQNDVexzHoOQ2iQmY/edit#heading=h.je8c7y5qpgki) fire that event when the user clicks on `<button type="submit">` in `<form>`. Our form components listen to that event to add their values to the `<form>`.
```

on a separate note, should we link to this page instead of the Google doc directly?

Reply:
Thanks @emyarod for reviewing! One question - What did you mean by "this page"? I wish there is a better explainer of the spec rather than a Google Doc, but there doesn't seem to be...

Reply:
@asudoh I meant to link this page https://www.chromestatus.com/feature/5662230242656256

Reply:
Got it, makes sense - Updated.